src/helpers/helpers.rtl.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+var getRtlAdapter = function(rectX, width) {
+	return {
+		x: function(x) {
+			return rectX + rectX + width - x;
+		},
+		setWidth: function(w) {
+			width = w;
+		},
+		textAlign: function(align) {
+			if (align === 'center') {
+				return align;
+			}
+			return align === 'right' ? 'left' : 'right';
+		},
+		xPlus: function(x, value) {
+			return x - value;
+		},
+		leftForLtr: function(x, itemWidth) {
+			return x - itemWidth;
+		},
+	};
+};
+
+var getLtrAdapter = function() {
+	return {
+		x: function(x) {
+			return x;
+		},
+		setWidth: function(w) { // eslint-disable-line no-unused-vars
+		},
+		textAlign: function(align) {
+			return align;
+		},
+		xPlus: function(x, value) {
+			return x + value;
+		},
+		leftForLtr: function(x, _itemWidth) { // eslint-disable-line no-unused-vars
+			return x;
+		},
+	};
+};
+
+var getAdapter = function(rtl, rectX, width) {
+	if (rtl) {
+		return getRtlAdapter(rectX, width);
+	}
+
+	return getLtrAdapter();
+};
+
+var overrideTextDirection = function(ctx, direction) {
+	delete ctx.prevTextDirection;
+
+	if (direction === 'ltr' || direction === 'rtl') {
+		var original = [
```

Comment:
Took some of it into consideration. Of course referencing `ctx.canvas.style` only later when necessary.
Btw, variables declared at the top of scope is an old C-style necessity, not necessary in higher level languages. When specifically using `let`/`const` in modern JS, you let the compiler do simple optimizations like allocating on the stack only when necessary.
I don't really get why Charts.js is set up with these antiquities...

Reply:
This project is still ES5 (for backward compatibility).

Declaring variables at top of scope helps minimization:  `var a,b,c;` vs `var a;(...)var b;(...)var c;`
I would have assigned cached `style` anyway, now its only saving couple of CPU cycles when input is invalid (probably only during some development) vs saving 2 bytes every time the package is downloaded (and the cycles in all that network equipment on the way).

No need to change anything though, just wanted to share my reasoning 🤔

Reply:
Well I wouldn't worry about 2 bytes :-)

Also my two cents are it's safe to assume that everyone is using some kind of transpiling (babel/rollup/webpack) pipeline, or at least mention that it is required and let people manage their own supported versions instead of limiting ourselves to lowest common ES :-)
This approach could actually save much more than 2 bytes (or 2 cpu cycles) for people targeting more modern platforms, and waste more for people targeting older platforms.